### PR TITLE
Cleanup pattern matching in parsing

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/pars.fsy [hunk 1]
```diff
@@ -136,17 +138,20 @@ let idOfPat m p =
     | _ -> raiseParseErrorAt m (FSComp.SR.parsIntegerForLoopRequiresSimpleIdentifier())
 
 let checkForMultipleAugmentations m a1 a2 = 
-    if not (isNil a1) && not (isNil a2) then raiseParseErrorAt m (FSComp.SR.parsOnlyOneWithAugmentationAllowed())
```
Comment (dsyme) [inline src/fsharp/pars.fsy]: clearer, shorter before - deal with errors first, then continue

Comment (forki) [reply src/fsharp/pars.fsy]: Good point.

**Comment by forki** [comment]
reverted that part.
done

**Comment by dsyme** [comment]
@forki I looked this over and don't find the code much clearer, so will close it if that's ok...

**Pull request closed without merging**